core/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -386,13 +386,24 @@ public static MetadataSnapshot readMetadataSnapshot(Path indexLocation, ESLogger
      * corruption markers.
      */
     public static boolean canOpenIndex(ESLogger logger, Path indexLocation) throws IOException {
+        try {
+            return tryOpenIndex(indexLocation);
+        } catch (Exception ex) {
+            logger.trace("Can't open index for path [{}]", ex, indexLocation);
+            return false;
+        }
+    }
+
+    /**
+     * Returns <code>true</code> iff the given location contains an index
+     * and the index can be successfully opened. If the index can not
+     * be opened, an exception is thrown
+     */
+    public static boolean tryOpenIndex(Path indexLocation) throws IOException {
```

Comment:
why do we need the return value? I doesn't add much as we throw an exception. I think it's clearer if we use void.